File: VB/WindowsApplication1/Custom Ribbon Control/MyRibbonPageGroupViewInfo.vb
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraBars.Ribbon.ViewInfo
Imports System.Drawing

Namespace WindowsApplication1
    Public Class MyRibbonPageGroupViewInfo
        Inherits RibbonPageGroupViewInfo

        Public Sub New(ByVal viewInfo As RibbonViewInfo, ByVal group As RibbonPageGroup)
            MyBase.New(viewInfo, group)
        End Sub
        Public Overrides Sub CalcViewInfo(ByVal bounds As Rectangle)
            If Object.Equals(PageGroup.Tag, "AlignRight") Then
                Dim offset As Integer = ViewInfo.Bounds.Right - bounds.Right
                If offset > 5 Then
                    bounds.Offset(offset, 0)
                End If
            End If
            MyBase.CalcViewInfo(bounds)
        End Sub
    End Class
End Namespace

File: CS/WindowsApplication1/Custom Ribbon Control/MyRibbonPageGroupViewInfo.cs
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraBars.Ribbon.ViewInfo;
using System.Drawing;

namespace WindowsApplication1 {
    public class MyRibbonPageGroupViewInfo : RibbonPageGroupViewInfo {
        public MyRibbonPageGroupViewInfo(RibbonViewInfo viewInfo, RibbonPageGroup group)
            : base(viewInfo, group) {
        }
        public override void CalcViewInfo(Rectangle bounds) {
            if (object.Equals(PageGroup.Tag, "AlignRight")) {
                int offset = ViewInfo.Bounds.Right - bounds.Right;
                if (offset > 5)
                    bounds.Offset(offset, 0);
            }
            base.CalcViewInfo(bounds);
        }
    }
}
